dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/EventExporterTest.java

```diff
@@ -592,23 +591,83 @@ void testExportEventsWhenFilteringByNumericDataElements()
   }
 
   @Test
-  void shouldFilterByEventsWithGivenDataValuesWhenFilterContainsDataElementUIDsOnly()
+  void shouldFilterByEventsContainingGivenDataValuesWhenFilteringByNonNullDataValues()
       throws ForbiddenException, BadRequestException {
     EventOperationParams params =
         EventOperationParams.builder()
+            .eventParams(EventParams.FALSE)
             .dataElementFilters(
                 Map.of(
                     UID.of("GieVkTxp4HH"),
-                    new ArrayList<>(),
+                    List.of(new QueryFilter(QueryOperator.NNULL)),
                     UID.of("GieVkTxp4HG"),
-                    new ArrayList<>()))
+                    List.of(new QueryFilter(QueryOperator.NNULL))))
             .build();
 
     List<String> events = getEvents(params);
 
     assertContainsOnly(List.of("kWjSezkXHVp"), events);
   }
 
+  @Test
+  void shouldFilterByEventsNotContainingGivenDataValueWhenFilteringByNullDataValues()
+      throws ForbiddenException, BadRequestException {
+    EventOperationParams params =
+        EventOperationParams.builder()
+            .enrollments(UID.of("nxP7UnKhomJ", "TvctPPhpD8z"))
+            .programStage(programStage)
+            .eventParams(EventParams.FALSE)
+            .dataElementFilters(
+                Map.of(UID.of("DATAEL00002"), List.of(new QueryFilter(QueryOperator.NULL))))
+            .build();
+
+    List<String> events = getEvents(params);
+
+    assertContainsOnly(List.of("pTzf9KYMk72"), events);
+  }
+
+  @Test
+  void shouldFilterByEventsContainingGivenDataValueWhenCombiningUnaryAndBinaryOperatorsInFilter()
+      throws ForbiddenException, BadRequestException {
+    DataElement dataElement = dataElement(UID.of("DATAEL00005"));
+    EventOperationParams params =
+        operationParamsBuilder
+            .enrollments(UID.of("nxP7UnKhomJ", "TvctPPhpD8z"))
+            .programStage(programStage)
+            .dataElementFilters(
+                Map.of(
+                    UID.of(dataElement),
+                    List.of(
+                        new QueryFilter(QueryOperator.IN, "option2"),
+                        new QueryFilter(QueryOperator.NNULL))))
+            .build();
+
+    List<String> events = getEvents(params);
+
+    assertContainsOnly(List.of("D9PbzJY8bJM"), events);
+  }
+
+  @Test
+  void shouldFilterByEventsContainingGivenDataValueWhenCombiningTwoUnaryOperatorsInFilter()
+      throws ForbiddenException, BadRequestException {
+    EventOperationParams params =
+        EventOperationParams.builder()
+            .enrollments(UID.of("nxP7UnKhomJ", "TvctPPhpD8z"))
+            .programStage(programStage)
+            .eventParams(EventParams.FALSE)
+            .dataElementFilters(
+                Map.of(
+                    UID.of("DATAEL00002"),
+                    List.of(
+                        new QueryFilter(QueryOperator.NNULL),
+                        new QueryFilter(QueryOperator.NNULL))))
```

Comment:
Why would we allow/support the same unary operator for the same data element?

Reply:
as discussed [here](https://dhis2.slack.com/archives/C05S1NTHBJ9/p1739176522707899?thread_ts=1738232925.732909&cid=C05S1NTHBJ9), that's what plaftorm does and we are adopting the same approach